tests/Filesystem.php

```diff
@@ -25,8 +25,20 @@ public static function readFile(string $fileName): string
     /**
      * @return list<string>
      */
-    public static function find(string $pattern): array
+    public static function findRecursively(string $path, string $pattern): array
     {
-        return glob($pattern) ?: [];
+        /**
+         * @var iterable<string, \SplFileInfo> $fileIterator
+         */
+        $fileIterator = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($path), \RecursiveIteratorIterator::CHILD_FIRST);
+
+        $found = [];
+        foreach ($fileIterator as $file) {
+            if (fnmatch($pattern, $file->getFilename())) {
```

Comment:
Obviously `glob` wasn't strict, but I think we're generally expecting this to only return files rather than directories. So it might be worth being explicit / concrete about that and checking `->isFile()` as well?

Reply:
the Finder-based implementation (replaced with `glob` in #297) was strict.

Reply:
An easy way to improve that is to add `\RecursiveIteratorIterator::LEAVES_ONLY` in the flags

Reply:
Good point - but this replaces a mix of different approaches to finding files (in the tests), others of which were just using `glob` and assuming there would only be files in the result.

Reply:
Both usages of this function seem to deal mainly with files, so I think I'll just rename the function and go for `isFile()`.